test/access/AccessControl.behavior.js

```diff
@@ -210,8 +211,293 @@ function shouldBehaveLikeAccessControlEnumerable(errorPrefix, admin, authorized,
+      const secondDefaultAdminTransferDelayedUntil = (await time.latest()).add(delay);
+      expect(await this.accessControl.pendingDefaultAdmin()).to.equal(other);
+      expect(await this.accessControl.defaultAdminTransferDelayedUntil()).to.be.bignumber.equal(
+        secondDefaultAdminTransferDelayedUntil,
+      );
+    });
+
+    it('should revert if it is called by non-admin accounts', async function () {
+      await expectRevert(
+        this.accessControl.beginDefaultAdminTransfer(newDefaultAdmin, { from: other }),
+        `${errorPrefix}: account ${other.toLowerCase()} is missing role ${DEFAULT_ADMIN_ROLE}`,
+      );
+    });
+  });
+
+  describe('accepts transfer admin', async function () {
+    let correctPendingDefaultAdmin;
+    let correctIncreaseTo;
+
+    beforeEach(async function () {
+      // Set as correct so they're explicitly needed in revert tests, since conditions
+      // are not mutually exclusive and accidentally an incorrect value will make it revert
+      // possiblty creating false positives (eg. expect revert for incorrect caller but getting it
+      // from a badly expected delayed until)
+      correctPendingDefaultAdmin = newDefaultAdmin;
+      correctIncreaseTo = (await time.latest()).add(delay).addn(1);
+
+      await this.accessControl.beginDefaultAdminTransfer(correctPendingDefaultAdmin, { from: defaultAdmin });
+    });
+
+    describe('caller is pending default admin and delayed until is met', async function () {
+      let from;
+
+      beforeEach(async function () {
+        await time.increaseTo(correctIncreaseTo);
+        from = correctPendingDefaultAdmin;
+      });
+
+      it('accepts a transfer and changes default admin', async function () {
+        await this.accessControl.acceptDefaultAdminTransfer({ from });
+        expect(await this.accessControl.hasRole(DEFAULT_ADMIN_ROLE, defaultAdmin)).to.be.false;
+        expect(await this.accessControl.hasRole(DEFAULT_ADMIN_ROLE, newDefaultAdmin)).to.be.true;
+        expect(await this.accessControl.owner()).to.equal(newDefaultAdmin);
+      });
+
+      it('accepts a transfer and emit events', async function () {
+        const receipt = await this.accessControl.acceptDefaultAdminTransfer({ from });
+        expectEvent(receipt, 'RoleRevoked', {
+          role: DEFAULT_ADMIN_ROLE,
+          account: defaultAdmin,
+        });
+        expectEvent(receipt, 'RoleGranted', {
+          role: DEFAULT_ADMIN_ROLE,
+          account: newDefaultAdmin,
+        });
+      });
+
+      it('accepts a transfer resetting pending default admin and delayed until', async function () {
+        await this.accessControl.acceptDefaultAdminTransfer({ from });
+        expect(await this.accessControl.defaultAdminTransferDelayedUntil()).to.be.bignumber.equal(web3.utils.toBN(0));
+        expect(await this.accessControl.pendingDefaultAdmin()).to.equal(ZERO_ADDRESS);
+      });
```

Comment:
We should put all of these in a single test IMO. @Amxx What do you think?

Reply:
Kept this one until the end so the commit can be easily reverted, but I unified them at https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4009/commits/a43abe7e27f82755ba2d3d47546a7bf6286cb4c7

It's consistent with what I did at https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4009/commits/ac093f42f13bbb6b52386c5c16212e3207d51ace